code/game/machinery/shieldgen.dm

```diff
@@ -247,6 +247,19 @@
 	var/obj/structure/cable/attached		// the attached cable
 	var/storedpower = 0
 
+/obj/machinery/shieldwallgen/science		//Use for xenobiology on maps where you have shield wall generators in xeno.
+	name = "shield wall generator"
+	desc = "A shield generator."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "Shield_Gen"
+	anchored = 0
+	density = 1
+	req_access = list(access_xenobiology)
```

Comment:
I could delete that, I didn't know how it worked so I thought I had to copy everything in. First time I did it it didn't compile because I pasted the variables too. (still doesn't compile, no clue why but I'll fix when I get home)